src/Nethermind/Nethermind.Core/Buffers/TinyArray.cs

```diff
@@ -0,0 +1,338 @@
+            differentBits |= Unsafe.ReadUnaligned<ulong>(ref Unsafe.Add(ref first, offset)) -
+                             Unsafe.ReadUnaligned<ulong>(ref Unsafe.Add(ref second, offset));
+
+            return differentBits == 0;
+        }
+
+        public static int MemorySize => Size;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    [InlineArray(Size)]
+    private struct Payload24 : IPayload
+    {
+        private byte _data;
+        private const int Size = 24;
+
+        public void SetSize(int size)
+        {
+            _data = (byte)size;
+        }
+
+        public void Load(ReadOnlySpan<byte> src)
+        {
+            _data = (byte)src.Length;
+            src.CopyTo(Span);
+        }
+
+        public byte Length => _data;
+
+        public Span<byte> Span => MemoryMarshal.CreateSpan(ref Unsafe.Add(ref _data, ValueOffset), _data);
+
+        public bool SequenceEqual(ReadOnlySpan<byte> other)
+        {
+            var length = Length;
+            if (length != other.Length)
+            {
+                return false;
+            }
+
+            ref byte first = ref Unsafe.Add(ref _data, ValueOffset);
+            ref byte second = ref MemoryMarshal.GetReference(other);
+
+            int offset = length - Vector128<byte>.Count;
+
+            return
+                ((Vector128.LoadUnsafe(ref first) ^ Vector128.LoadUnsafe(ref second)) |
+                 (Vector128.LoadUnsafe(ref Unsafe.Add(ref first, offset)) ^
+                  Vector128.LoadUnsafe(ref Unsafe.Add(ref second, offset)))) ==
+                Vector128<byte>.Zero;
+        }
+
+        public static int MemorySize => Size;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    [InlineArray(Size)]
+    private struct Payload32 : IPayload
+    {
+        private byte _data;
+        private const int Size = 32;
+
+        public void SetSize(int size)
+        {
+            _data = (byte)size;
+        }
+
+        public void Load(ReadOnlySpan<byte> src)
+        {
+            _data = (byte)src.Length;
+            src.CopyTo(Span);
+        }
+
+        public byte Length => _data;
+
+        public Span<byte> Span => MemoryMarshal.CreateSpan(ref Unsafe.Add(ref _data, ValueOffset), _data);
+
+        public bool SequenceEqual(ReadOnlySpan<byte> other)
+        {
+            var length = Length;
+            if (length != other.Length)
+            {
+                return false;
+            }
+
+            ref byte first = ref Unsafe.Add(ref _data, ValueOffset);
+            ref byte second = ref MemoryMarshal.GetReference(other);
+
+            int offset = length - Vector128<byte>.Count;
+
+            return
+                ((Vector128.LoadUnsafe(ref first) ^ Vector128.LoadUnsafe(ref second)) |
+                 (Vector128.LoadUnsafe(ref Unsafe.Add(ref first, offset)) ^
+                  Vector128.LoadUnsafe(ref Unsafe.Add(ref second, offset)))) ==
+                Vector128<byte>.Zero;
```

Comment:
Why Vector128 and not 256 or 512 for example?

Reply:
Totally removed.